### ISSUE TITLE
program.start() executions need monitoring for a clear "ready" state

### ISSUE THREAD

**Issue by amauchley** [PSLCL/testing-framework#188]
Multiple CST tests are failing due to timeouts on the requestor client side.  Where CST servers are up indefinitely the provider (program.start()) and the requestor (program.run()) application is able to connect into the CST server fairly quick, causing the requestor to fail it's waitProvide since the provider isn't fully up and ready yet.  In the past we have worked around this issue by just including multiple attempts and longer timeouts, but this also can cause a false success for tests if they eventually receive the provide after multiple attempts. Also, it may cause a long time for the test execution to finally finish with a failure if there really is a bug in the application under test, and it has to go through all of the timeouts.  There needs to be monitoring on any program.start executions in order to notify the dtfrunner that the dependent program.run should execute so that the test race condition doesn't happen.

**Comment by Chad-Adams** [comment]
Is not DOF client/provider connectivity functionality all about solving just what you state?  i.e. is not the client side waitProvide intended to handle all possible provider side states of availability?

"application is able to connect into the CST server fairly quick, causing the requestor to fail it's waitProvide since the provider isn't fully up "

sounds like a client side waitProvide bug or the provider is never actually coming up?

**Comment by jwclark** [comment]
Agreed that it sounds like a bug in the CST if the provide operation doesn't eventually show up unless you retry the waitProvide, or if a provide is received when not expected after multiple attempts.

Working on some ideas for how this could be accomplished.
